Core/forms.py

```diff
@@ -165,3 +169,27 @@ def __init__(self, *args, **kwargs):
                 'type': 'checkbox'}
             )
         )
+
+    def clean(self):
+        registered = None
+        email_exist = None
+        student_number = self.cleaned_data['student_number']
+        email = self.cleaned_data['email']
+        student_number = int(student_number)
+        event = Event.objects.filter(active=True).first()
+        try:
+            registered = Registration.objects.filter(student_number=student_number, event=event)
+        except:
+            return student_number
+
+        try:
```

Comment:
Again, incorrect use of `try...except` as the exception would never be raised.
A solution can be if you want `email` and `event` to be unique together, then add them in the model only and it will be checked in the call to `super`.